test/arithmetic/floating_point/floating_point_sqrt_test.dart

```diff
@@ -0,0 +1,200 @@
+        final eOut = sqrtDUT.error;
+
+        expect(eOut.value, equals(expError.value),
+            reason: 'error =\n'
+                '\t${eOut.value} actual\n'
+                '\t${expError.value} expected');
+      }
+    }
+  });
+
+  test('FP: targeted normalized sqrt', () {
+    const exponentWidth = 8;
+    const mantissaWidth = 23;
+
+    final fp = FloatingPoint(
+        exponentWidth: exponentWidth, mantissaWidth: mantissaWidth);
+
+    for (final sqrtDUT in [
+      FloatingPointSqrtSimple(fp),
+    ]) {
+      final testCases = [
+        144.0,
+        288.0,
+        3.567,
+        1123.5,
+        17.0,
+        92.5,
+        100.0,
+        85.672,
+      ];
+
+      for (final test in testCases) {
+        final fv = FloatingPointValue.populator(
+                exponentWidth: exponentWidth, mantissaWidth: mantissaWidth)
+            .ofDouble(test);
+
+        fp.put(fv);
+
+        final compResult = sqrtDUT.sqrtR;
+        final compError = sqrtDUT.error;
+
+        final expResult = FloatingPointValue.populator(
+                exponentWidth: exponentWidth, mantissaWidth: mantissaWidth)
+            .ofDouble(sqrt(test));
+        final expError = Const(0);
+        expect(compResult.floatingPointValue.toDouble(),
+            equals(expResult.toDouble()),
+            reason: '\t${fp.floatingPointValue} '
+                '(${fp.floatingPointValue.toDouble()}) =\n'
+                '\t${compResult.floatingPointValue}'
+                '(${compResult.floatingPointValue.toDouble()}) actual\n'
+                '\t$expResult (${expResult.toDouble()}) expected');
+
+        expect(compError.value, equals(expError.value),
+            reason: 'error =\n'
+                '\t${expError.value} actual\n'
+                '\t${expError.value} expected');
+      }
+    }
+  });
+
+  test('FP: random number sqrt', () {
+    const exponentWidth = 5;
+    const mantissaWidth = 9;
+
+    final fp = FloatingPoint(
+        exponentWidth: exponentWidth, mantissaWidth: mantissaWidth);
+
+    final sqrtDUT = FloatingPointSqrtSimple(fp);
+    final rand = Random(513);
+    for (var i = 0; i < 4096; i++) {
+      final fv = FloatingPointValue.populator(
+              exponentWidth: exponentWidth, mantissaWidth: mantissaWidth)
+          .random(rand, normal: true);
+      fp.put(fv);
+      // only want to test on positive real values
+      if (fp.isAnInfinity.value.toBool() ||
+          fp.isNaN.value.toBool() ||
+          fp.isAZero.value.toBool() ||
+          fp.sign.value.toBool()) {
+        continue;
+      }
+      final compResult = sqrtDUT.sqrtR;
+      final compError = sqrtDUT.error;
+
+      final expResult = FloatingPointValue.populator(
+              exponentWidth: exponentWidth, mantissaWidth: mantissaWidth)
+          .ofDouble(sqrt(fv.toDouble()));
+      final expError = Const(0);
+
+      expect(compResult.floatingPointValue.withinRounding(expResult), true,
+          reason: '\t${fp.floatingPointValue} '
```

Comment:
Here's an opportunity to use 
```
reason:   '''
${fp.floatingPointValue} 
<avoiding having to inject newlines into a string>
'''
);

Reply:
I'm not sure I know what you are asking here @desmonddak

Reply:
You should use the Dart multi-line string to avoid having to add newlines, tabs, or extra quotes in your strings.  You can format the string exactly how you want it in your code without having to specify tabs or newlines or quotes.  Just start with three single quotes, then the lines after that are your string, then end with three single quotes.